### ISSUE TITLE
Invalid YAML references within string values

### ISSUE THREAD

**Issue by matteblair** [tangrams/outdoor-style#55]
https://github.com/tangrams/outdoor-style/blob/gh-pages/outdoor-style.yaml#L4594
https://github.com/tangrams/outdoor-style/blob/gh-pages/outdoor-style.yaml#L4689

Referencing a YAML anchor from within a string (such as a JS function) will not substitute the correct value for the anchor. In the lines above, those function strings are interpreted verbatim and produce a JS syntax error. 

To see how YAML anchors are resolved (or not resolved), try pasting this:

``` yaml
Define anchor: &foo Bar
Reference anchor: *foo
Interpolate anchor: Snickers *foo
```

into this:
http://www.yamllint.com/

To correct this in the lines above, I believe you can add the zoom constraint to the `filter` immediately above each line and use the anchor value directly in the `visible` node:

``` yaml
filter: { area: false, min_zoom: { min: $zoom } }
...
      visible: *icon_visible_poi_landuse
```

Incidentally, filter statements are _much_ more efficient than JS functions, I recommend using them wherever possible.

**Comment by nvkelso** [comment]
Since the file has changed since the issue was reported:
- https://github.com/tangrams/outdoor-style/blob/0256dc198c2025ec19300c50c5345574aa313fe4/outdoor-style.yaml#L4594
- https://github.com/tangrams/outdoor-style/blob/0256dc198c2025ec19300c50c5345574aa313fe4/outdoor-style.yaml#L4689

But the I get errors:

```
            direct-match:
               filter: { area: false, min_zoom: { min: $zoom } }
               draw:
                   icons:
                       # sprite_source: kind
                       sprite: function() { return feature.kind; }
                       visible: *icon_visible_poi_landuse
```

Tangram v0.8.0 [error]: tile load error for mapzen/16/10482/25330/16: ReferenceError: $zoom is not defined at: ReferenceError: $zoom is not defined.

**Comment by nvkelso** [comment]
Related: #39.

**Comment by nvkelso** [comment]
Looks like there were 3 of these, the last one being:

```
            car-parking:
                filter: { kind: [parking], $zoom: { min: 17 } }
                draw:
                    icons:
                        sprite: parking
                        visible: function() { if( feature.min_zoom <= $zoom ) { return *icon_visible_poi_landuse_e; } else { return false; } }
```

**Comment by matteblair** [comment]
Oh, yep I was mistaken about that filter syntax, we can't currently do that (a little bit unexpected, but enabling that would be tricky). Instead we would need something like:

``` yaml
filter: 
  all:
    - function() { return feature.min_zoom <= $zoom }
    - { kind: parking, $zoom: { min: 17 } } 
```

That should let us use just the anchor value for the `visible` node.

**Comment by nvkelso** [comment]
Fixed in 24f014db5356911180c3d1da34a2bca7c6be57f6 and b71bec7230b45c339c464ae9fab7f508be1330e2.
